pyxem/generators/indexation_generator.py

```diff
@@ -34,35 +31,48 @@
 
 def correlate_library(image, library, n_largest, mask, keys=[]):
     """Correlates all simulated diffraction templates in a DiffractionLibrary
-    with a particular experimental diffraction pattern (image) stored as a
-    numpy array. See the correlate method of IndexationGenerator for details.
+    with a particular experimental diffraction pattern (image).
+
+    Parameters
+    ----------
+    image : np.array()
+        The experimental diffraction pattern of interest.
+    library : DiffractionLibrary
+        The library of diffraction simulations to be correlated with the
+        experimental data.
+    n_largest : int
+        The number of well correlated simulations to be retained.
+    mask : bool array
+        A mask for navigation axes 1 indicates positions to be indexed.
+
+    Returns
+    -------
+    top_matches : (<num phases>*n_largest, 5), np.array()
+        A numpy array containing the top n correlated simulations for the
+        experimental pattern of interest.
+
+    See also
+    --------
+    pyxem.utils.correlate and the correlate method of IndexationGenerator.
     """
-    i = 0
-    out_arr = np.zeros((n_largest * len(library), 5))
+    top_matches = np.zeros((len(library), n_largest, 5))
     if mask == 1:
-        for key in library.keys():
-            correlations = dict()
-            for orientation, diffraction_pattern in library[key].items():
-                # diffraction_pattern here is in fact a library of
-                # diffraction_pattern_properties
+        for phase_index, key in enumerate(library.keys()):
```

Comment:
Here I think `enumerate` is also worth commenting on. Because `library.keys()` could give any ordering it means you won't get reproducible results. Something like

```
if phase_labels:
     phase_index = phase_labels[key]
```

underneath would help (`phase_label` would be dict() that gets passed instead of the old `keys` )

Reply:
I don't think this is a big deal - or should be solved by passing keys through objects, which will be covered in #315